### PR TITLE
[docs] Clarifying that SQL doesn't support recursive queries yet

### PR DESCRIPTION
**Pull request by gunnarmorling**
Quick draft for the doc change discussed on the mailing list here: https://lists.apache.org/thread/f6zcfsdh5f9dhnknmn2rr4qbf7whycsp

Feedback welcome, happy to adjust this change as needed. E.g. does it need a Jira issue? Thanks!

### CODE REVIEW

File: site/_docs/algebra.md [hunk 1]
```diff
@@ -309,6 +299,19 @@ LogicalRepeatUnion(all=[true])
         LogicalTableScan(table=[[aux]])
 {% endhighlight %}
 
+Note that currently there is no support for recursive queries in the SQL layer yet.
```
Comment (zabetak) [inline site/_docs/algebra.md:302]: "currently" and "yet" seem a bit superfluous; remove one of them.

File: site/_docs/algebra.md [hunk 2]
```diff
@@ -265,17 +265,7 @@ or `builder.field(2, 1, 1)`.
...
-{% highlight sql %}
-WITH RECURSIVE aux(i) AS (
-  VALUES (1)
-  UNION ALL
-  SELECT i+1 FROM aux WHERE i < 10
-)
-SELECT * FROM aux
-{% endhighlight %}
-
+A recursive query, e.g. this one that generates the sequence 1, 2, 3, ...10,
```
Comment (zabetak) [inline site/_docs/algebra.md:268]: I prefer the SQL example here as I find it easier to follow the example afterwards.

**Comment by rubenada** [comment]
Thanks for the PR @gunnarmorling .

Instead of using "[docs]" as prefix for the PR title and commit message, we usually use a de-facto convention of "Site: ...". And I'd prefer to avoid contractions in there too.
So I'd propose something like: "Site: Clarifying that SQL does not support recursive queries yet" for PR title and final commit message after squashing all potential commits into a single one.

File: site/_docs/algebra.md [hunk 3]
```diff
@@ -309,6 +299,19 @@ LogicalRepeatUnion(all=[true])
...
+
+{% highlight sql %}
+WITH RECURSIVE aux(i) AS (
+  VALUES (1)
+  UNION ALL
+  SELECT i+1 FROM aux WHERE i < 10
+)
+SELECT * FROM aux
+{% endhighlight %}
+
```
Comment (zabetak) [inline site/_docs/algebra.md:314]: Maybe we can keep the note and here just write: "The SQL example above is only for illustrative purposes (`RECURSIVE` keyword not supported)."

File: site/_docs/algebra.md (hunk 2 above)
Comment (rubenada) [inline site/_docs/algebra.md:268]: nitpick: with the change, IMO the expression "this one" sounds a bit strange. Proposal:
```
A recursive query, e.g. generating the sequence 1, 2, 3, ...10,
can be produced using a scan on a TransientTable and a RepeatUnion:
```

**Review by zabetak — APPROVED** [review]
Minor suggestions based on personal preferences.

File: site/_docs/algebra.md (hunk 2 above)
Comment (rubenada) [reply site/_docs/algebra.md:268]: UPDATE: I have checked @zabetak  comments, and I agree, I think it is clearer to keep the SQL example first. 
So I'll resolve this conversation, my comment can be disregarded in favor of @zabetak 's suggestion.

**Comment by gunnarmorling** [comment]
Excellent, thanks a lot!

**Pull request closed without merging**